mantidproject/mantid
# Output Build Logs to a File for the Jenkins Warnings Plugin

### Description of work

#### Summary of work
Added a file output in the same style as the `ctest` log for the build log. 

Once this is merged, the configurations for the 3 PR OSs need to be changed to check this log when the `Warnings` plugin is run. 

- [x] https://builds.mantidproject.org/job/pull_requests-conda-linux/configure > Post-build Actions
- [x] https://builds.mantidproject.org/job/pull_requests-conda-osx/configure > Post-build Actions
- [x] https://builds.mantidproject.org/job/pull_requests-conda-windows/configure > Post-build Actions

Fixes #38148


#### Further detail of work


### To test:
1. Rerun a job (or run the build script locally). There should be a `.log` file associated with each OS containing the log from the build step. 


*This does not require release notes* because **it is not a user-facing change.**

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
